src/Passes/Source/Generators/LlvmPassesConfig.hpp

```diff
@@ -0,0 +1,29 @@
+#pragma once
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "Commandline/ConfigurationManager.hpp"
+
+namespace microsoft
+{
+namespace quantum
+{
+
+    class LlvmPassesConfiguration
```

Comment:
It might be worthwhile to think about how defining a configuration for each pass would look like and whether a common interface for doing so would make sense. It might make sense to have a per-pass pieces for each pass as part of a common configuration for all passes.

Reply:
That is already done. This setup is a per-piece configuration and you will notice that each profile component (which may consist of one or more passes) have their own config. See for instance component registration in `DefaultProfileGenerator`:

```c++
        registerProfileComponent<RuleTransformationPassConfiguration>(
            "transformation-rules",
            [this](RuleTransformationPassConfiguration const& config, IProfileGenerator* ptr, Profile&) {
                auto& ret = ptr->modulePassManager();

                // Defining the mapping
                RuleSet rule_set;
                auto    factory = RuleFactory(rule_set);
                factory.usingConfiguration(configurationManager().get<FactoryConfiguration>());

                // Creating profile pass
                ret.addPass(RuleTransformationPass(std::move(rule_set), config));
            });

        registerProfileComponent<LlvmPassesConfiguration>(
            "llvm-passes", [](LlvmPassesConfiguration const& config, IProfileGenerator* ptr, Profile&) {
                // Configuring LLVM passes
                if (config.alwaysInline())
                {
                    auto& ret          = ptr->modulePassManager();
                    auto& pass_builder = ptr->passBuilder();
                    ret.addPass(llvm::AlwaysInlinerPass());

                    auto inliner_pass = pass_builder.buildInlinerPipeline(
                        ptr->optimisationLevel(), llvm::PassBuilder::ThinLTOPhase::None, ptr->debug());
                    ret.addPass(std::move(inliner_pass));
                }
            });
```
Whether or not we choose that a component _always_ must be comprised of exactly one pass is a different discussion.